### PR TITLE
Change_Perspective_branch

### PR DESCRIPTION
**Pull request by 0luwatamilore**
New Commit: I added a button to the action bar which enables the user to change his view perspective.

First perspective shows a single video per row, Second perspective shows two videos per row, Third perspective shows three videos per row.
Created a ViewType class to store perspective types.
Replaced my bottom navigation view with a tab layout for easy navigation.
Implemented a swipe navigation feature.
The swipe Navigation feature enables the user to move between fragments and tabs by swiping on the screen.

### CODE REVIEW

**Review by ragrawal4545 — APPROVED** [review]
lgtm!

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -7,27 +7,45 @@
...
         android:layout_height="wrap_content"
+        android:layout_alignParentBottom="true"
         android:layout_alignParentStart="true"
         android:layout_alignParentEnd="true"
-        android:layout_alignParentBottom="true"
-        app:menu="@menu/menu_bottom_navigation" />
+        />
+
+    <!--    <FrameLayout-->
+<!--        android:id="@+id/flContainer"-->
```
Comment (0luwatamilore) [inline app/src/main/res/layout/activity_main.xml]: I saw the dead code and have removed them in my latest commit

File: app/src/main/java/com/example/news_app/HistoryAdapter.java [hunk 2]
```diff
@@ -17,8 +17,8 @@
 
 public class HistoryAdapter extends RecyclerView.Adapter<HistoryAdapter.ViewHolder> {
 
-    private Context context;
-    private List<History> allHistory;
+    private final Context context;
```
Comment (NateWickstrom) [inline app/src/main/java/com/example/news_app/HistoryAdapter.java:20]: nice!

File: app/src/main/java/com/example/news_app/MainActivity.java [hunk 3]
```diff
@@ -2,66 +2,63 @@
...
-                    fragment = new ProfileFragment();
-                    break;
+    public  class ViewPagerFragmentStateAdapter extends FragmentStateAdapter {
+        public ViewPagerFragmentStateAdapter(FragmentActivity fragmentActivity) {
+            super(fragmentActivity);
+        }
+        @Override
+        public Fragment createFragment(int position) {
+            switch (position) {
+                case 0:
```
Comment (NateWickstrom) [inline app/src/main/java/com/example/news_app/MainActivity.java:45]: Magic number:
https://refactoring.guru/replace-magic-number-with-symbolic-constant

File: app/src/main/java/com/example/news_app/VideoAdapter.java [hunk 4]
```diff
@@ -23,19 +23,31 @@ public class VideoAdapter extends RecyclerView.Adapter<VideoAdapter.ViewHolder>
...
+        this.viewType= viewType;
     }
 
-
     @NonNull
     @Override
     public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View videoView = null;
+        switch (this.viewType) {
+            case OneColumn:
```
Comment (NateWickstrom) [inline app/src/main/java/com/example/news_app/VideoAdapter.java:40]: naming convention for enum values should be all caps
https://stackoverflow.com/a/3069863

File: app/src/main/java/com/example/news_app/fragment/HomeFragment.java [hunk 5]
```diff
@@ -4,56 +4,54 @@
...
+    private ViewType selectedViewType = ViewType.OneColumn;
     public static final String TAG = "HOME FRAGMENT";
-    String secretValue;
-    String NOW_PLAYING_URL;
     private RecyclerView rvVideos;
     private VideoAdapter adapter;
     private SwipeRefreshLayout swipeContainer;
+    private Menu menu;
     List<Video> allVideos;
+    String NOW_PLAYING_URL;
```
Comment (NateWickstrom) [inline app/src/main/java/com/example/news_app/fragment/HomeFragment.java:53]: only static final class values should be all caps

File: app/src/main/res/layout/item_video_two_columns.xml [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        <TextView
+            android:id="@+id/tvSeparator2"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_below="@+id/tvTitle"
+            android:layout_marginStart="5dp"
+            android:layout_marginTop="0dp"
+            android:layout_toEndOf="@+id/tvPublishedAt"
+            android:text="#"
+            android:textSize="8sp"/>
```
Comment (NateWickstrom) [inline app/src/main/res/layout/item_video_two_columns.xml:75]: better to use Android Typography type
https://developer.android.com/reference/kotlin/androidx/compose/material/Typography

**Review by NateWickstrom — COMMENTED** [review]
looks good - just some minor formating

**Pull request merged**